src/librustc_mir/interpret/snapshot.rs

```diff
@@ -291,6 +335,28 @@ struct FrameSnapshot<'a, 'tcx: 'a> {
     stmt: usize,
 }
 
+// Not using the macro because that does not support types depending on 'tcx
```

Comment:
That's surprising, I'd expect it's because `'mir`, not `'tcx` - lots of types depend on `'tcx`.

Reply:
> When can we `#[derive]` this? :D

https://github.com/rust-lang/rust/pull/49219, which is lower priority than the edition :(